iothub/device/src/Common/Singleton.cs

```diff
@@ -45,14 +45,22 @@ public Task OpenAsync(TimeSpan timeout, CancellationToken cancellationToken)
 
         public Task CloseAsync()
         {
-            this.Dispose();
+            var thisTaskCompletionSource = this.taskCompletionSource;
+            if (thisTaskCompletionSource != null && thisTaskCompletionSource.Task.Status == TaskStatus.RanToCompletion)
+            {
+                OnSafeClose(thisTaskCompletionSource.Task.Result);
+            }
 
             return TaskHelpers.CompletedTask;
         }
 
         public void Close()
         {
-            this.Dispose();
+            var thisTaskCompletionSource = this.taskCompletionSource;
+            if (thisTaskCompletionSource != null && thisTaskCompletionSource.Task.Status == TaskStatus.RanToCompletion)
+            {
+                OnSafeClose(thisTaskCompletionSource.Task.Result);
```

Comment:
I'm not sure I understand this flow: when FaultTolerantAmqpObject gets OnSafeClose it'll call this.closeObject.

What is the lifetime expectation for this object? Who will dispose it (I don't see any dispose calls in the source code).
I do see Close() is called within the IotHubSingleTokenConnection.Release.